src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -464,207 +537,246 @@ private void onCreateChallenge() {
-		actionMenu.setActionListener(Action.DELETE_ENTITY, this);
+		if (isTopLevelProjectToolsMenu(entityBundle.getEntity(), currentArea)) {
+			actionMenu.setActionVisible(Action.DELETE_ENTITY, false);
+		} else {
+			actionMenu.setActionVisible(Action.DELETE_ENTITY, permissions.getCanDelete());
+			actionMenu.setActionText(Action.DELETE_ENTITY, DELETE_PREFIX+enityTypeDisplay);
+			actionMenu.setActionListener(Action.DELETE_ENTITY, this);
+		}
 	}
 	private void configureDeleteWikiAction(){
-		if(entityBundle.getEntity() instanceof Project){
+		if(isWikiableConfig(entityBundle.getEntity(), currentArea) && 
+				entityBundle.getEntity() instanceof Project){
 			actionMenu.setActionVisible(Action.DELETE_WIKI_PAGE, permissions.getCanDelete());
-			actionMenu.setActionEnabled(Action.DELETE_WIKI_PAGE, permissions.getCanDelete());
 			actionMenu.setActionListener(Action.DELETE_WIKI_PAGE, this);
 		} else {
 			actionMenu.setActionVisible(Action.DELETE_WIKI_PAGE, false);
-			actionMenu.setActionEnabled(Action.DELETE_WIKI_PAGE, false);
 		}
 	}
 	
-	
 	private void configureShareAction(){
-		actionMenu.setActionEnabled(Action.SHARE, true);
-		actionMenu.setActionVisible(Action.SHARE, true);
-		actionMenu.setActionListener(Action.SHARE, this);
-		GlobalApplicationState globalAppState = getGlobalApplicationState();
-		if(PublicPrivateBadge.isPublic(entityBundle.getBenefactorAcl(), globalAppState.getPublicPrincipalIds())){
-			actionMenu.setActionIcon(Action.SHARE, IconType.GLOBE);
-		}else{
-			actionMenu.setActionIcon(Action.SHARE, IconType.LOCK);
+		if (isTopLevelProjectToolsMenu(entityBundle.getEntity(), currentArea)) {
+			actionMenu.setActionVisible(Action.SHARE, false);
+		} else {
+			actionMenu.setActionVisible(Action.SHARE, true);
+			actionMenu.setActionListener(Action.SHARE, this);
+			actionMenu.setActionText(Action.SHARE, enityTypeDisplay + " Sharing Settings");
+			GlobalApplicationState globalAppState = getGlobalApplicationState();
+			if(PublicPrivateBadge.isPublic(entityBundle.getBenefactorAcl(), globalAppState.getPublicPrincipalIds())){
+				actionMenu.setActionIcon(Action.SHARE, IconType.GLOBE);
+			}else{
+				actionMenu.setActionIcon(Action.SHARE, IconType.LOCK);
+			}	
 		}
 	}
 	
+	public boolean isTopLevelProjectToolsMenu(Entity entity, EntityArea area) {
+		return entity instanceof Project && area != null;
+	}
+	
+	public boolean isWikiableConfig(Entity entity, EntityArea area) {
+		if (entity instanceof Project) {
+			return EntityArea.WIKI.equals(area);
+		} 
+		return isWikiableType(entity);
+	}
+	
+	public boolean isContainerOnFilesTab(Entity entity, EntityArea area) {
+		return (entity instanceof Folder || 
+				(entity instanceof Project && EntityArea.FILES.equals(area)));
+	}
+	
 	/**
 	 * Can an entity of this type be moved?
 	 * @param entity
 	 * @return
 	 */
 	public boolean isMovableType(Entity entity){
-		if(entity instanceof Project){
+		if(entity instanceof Project || entity instanceof DockerRepository){

```

Comment:
is this true?  suggest remove

Reply:
No, docker repo widget must have controlled it's action menu.  Move is not allowed for docker repo entities (on prod), so keep this change.